Add back CMAKE_OSX_SYSROOT to documented build steps on macOS

# Bug report

### What operating system and version are you using?

macOS 26.5.2

### What version of osquery are you using?

version = 5.23.0-29-g99239193e

### What steps did you take to reproduce the issue?

```bash
mkdir build; cd build
cmake ..
cmake --build . -j $(sysctl -n hw.ncpu)
```

Which is currently what the wiki says to do (as of yesterday).

### What did you expect to see?

`[100%] Built target create_osqueryi`

### What did you see instead?

```text
[ 18%] Building C object libs/src/libarchive/CMakeFiles/thirdparty_libarchive.dir/src/libarchive/archive_read_support_filter_compress.c.o
[ 18%] Built target osquery_distributed_ns_gen_1
In file included from apps/lib/app_libctx.c:9:
In file included from apps/include/app_libctx.h:13:
In file included from include/openssl/types.h:32:
include/openssl/e_os2.h:203:10: fatal error: 'sys/types.h' file not found
  203 | #include <sys/types.h>
      |          ^~~~~~~~~~~~~
```

This, on the other hand, _works_:

```bash
cmake -DCMAKE_OSX_SYSROOT=$(xcrun --sdk macosx --show-sdk-path) ..
cmake --build . -j $(sysctl -n hw.ncpu)
```

Maybe yesterday's change was incomplete? 

https://github.com/osquery/osquery/commit/7f1b8bb30d9200823c53dba4f5488182da7b079e#diff-287cd509d7f4468dc64f8a2e1c65e05e41b1308759a190f490c4de92f8006c69R52-L90

--- a/cmake/toolchain.cmake
+++ b/cmake/toolchain.cmake
@@ -19,7 +19,7 @@ else()
 endif()
 
 
-if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Darwin")
   if(NOT DEFINED CMAKE_OSX_SYSROOT OR "${CMAKE_OSX_SYSROOT}" STREQUAL "")
     execute_process(
       COMMAND xcrun --sdk macosx --show-sdk-path

